scripts/common.sh

```diff
@@ -216,6 +216,68 @@ gcloud_auth() {
     gcloud auth login --update-adc
 }
 
+ensure_tf_backend() {
+  GCP_CREDENTIALS_PATH="${1}"
+  shift
+  TF_STATA_BUCKET_NAME="gs://tf-state-${DEFAULT_PROJECT}/"
+  gcloud_exec_cmds "${GCP_CREDENTIALS_PATH}" "$(pwd)" "gsutil list -p ${DEFAULT_PROJECT} | grep ${TF_STATA_BUCKET_NAME}"
+  if [ -z "${DOCKER_RUN_OUTPUT}" ]; then
+    echo "Terraform backend storage does not exists, creating..."
+    gcloud_exec_cmds "${GOOGLE_APPLICATION_CREDENTIALS_PATH}" "$(pwd)" "gsutil mb -p ${DEFAULT_PROJECT} --pap enforced -b on -l ${DEFAULT_REGION} ${TF_STATA_BUCKET_NAME}"
+  else
+    echo "Terraform backend exists, skip..."
+  fi
+
+  # If the bucket still not exists
+  gcloud_exec_cmds "${GCP_CREDENTIALS_PATH}" "$(pwd)" "gsutil list -p ${DEFAULT_PROJECT} | grep ${TF_STATA_BUCKET_NAME}"
+  if [ -z "${DOCKER_RUN_OUTPUT}" ]; then
```

Comment:
Why don't you check for the return code instead of checking the output? That would be the idiomatic way.

Reply:
refactored this.